src/System.Private.ServiceModel/tests/Scenarios/Client/ClientBase/ClientBaseTests.cs

```diff
@@ -125,31 +125,74 @@ public static void ClientMessageInspector_Verify_Invoke()
 
     [Fact]
     [OuterLoop]
-    public static void ClientBaseOfT_Call()
+    public static void ClientBaseOfT_HttpTransport_Call()
     {
-        // This test verifies ClientBase<T> can be used to create a proxy and invoke an operation
-        StringBuilder errorBuilder = new StringBuilder();
+        // This test verifies ClientBase<T> can be used to create a proxy and invoke an operation over Http
+
+        CustomBinding customBinding = new CustomBinding();
+        customBinding.Elements.Add(new TextMessageEncodingBindingElement());
+        customBinding.Elements.Add(new HttpTransportBindingElement());
+
+        MyClientBase client = new MyClientBase(customBinding, new EndpointAddress(Endpoints.HttpSoap12_Address));
+        Assert.Equal(CommunicationState.Created, client.State);
+
+        IWcfServiceGenerated serviceProxy = client.ChannelFactory.CreateChannel();
+        Assert.Equal(CommunicationState.Opened, client.State);
+
         try
         {
-            CustomBinding customBinding = new CustomBinding();
-            customBinding.Elements.Add(new TextMessageEncodingBindingElement());
-            customBinding.Elements.Add(new HttpTransportBindingElement());
+            string result = serviceProxy.Echo("Hello");
+            Assert.Equal("Hello", result);
 
-            MyClientBase client = new MyClientBase(customBinding, new EndpointAddress(BaseAddress.HttpBaseAddress));
-            IWcfServiceGenerated serviceProxy = client.ChannelFactory.CreateChannel();
+            Assert.Equal(CommunicationState.Opened, client.State);
 
-            string result = serviceProxy.Echo("Hello");
-            if (!string.Equals(result, "Hello"))
+            client.Close();
```

Comment:
Close() is not in the public contract.  This test fails to compile in our CoreCLR/NET Native test environment.